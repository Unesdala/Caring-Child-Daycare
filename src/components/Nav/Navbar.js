import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import ReactResizeDetector from 'react-resize-detector';
import DefaultCollapsableNav from './CollapsableNav';

export default class Navigation extends Component {
  constructor(props) {
    super(props);
    this.parentRef = React.createRef();
    this.onResize = this.onResize.bind(this);
    this.state = { width: 320 };
  }

  onResize(width) {
    this.setState({ width });
  }

  render() {
    const { width } = this.state;
    return (
      <>
        {width <= 768 - 18 || width === 751
          ? (
            <DefaultCollapsableNav />
          ) : (
            <nav className="menu">
              <ul className="menu-nav">
                <li className="nav-item"><NavLink to="/daycare" className="nav-link" exact activeClassName="active">Home</NavLink></li>
                <li className="nav-item"><NavLink to="/daycare/centers" className="nav-link">Our Centers</NavLink></li>
                <li className="nav-item"><NavLink to="/daycare/about" className="nav-link">About Us</NavLink></li>
                <li className="nav-item"><NavLink to="/daycare/programs" className="nav-link">Programs</NavLink></li>
                <li className="nav-item"><NavLink to="/daycare/parents" className="nav-link">Parent&apos;s Corner</NavLink></li>
                <li className="nav-item"><NavLink to="/daycare/news" className="nav-link">News</NavLink></li>
                <li className="nav-item"><NavLink to="/daycare/careers" className="nav-link">Careers</NavLink></li>
                <li className="nav-item"><NavLink to="/daycare/contact" className="nav-link">Contact Us</NavLink></li>
              </ul>
            </nav>
          )}
        <ReactResizeDetector handleWidth handleHeight onResize={this.onResize} targetDomEl={this.parentRef.current} />
      </>
    );
  }
}
