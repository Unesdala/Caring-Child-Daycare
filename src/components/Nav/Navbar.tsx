import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import ReactResizeDetector from 'react-resize-detector';
import DefaultCollapsableNav from './CollapsableNav';
import DefaultParentsNav, { ProgramsNav, AboutNav } from './DropdownNav';

interface NavigationState {
  width: number;
}

export default class Navigation extends Component<Record<string, unknown>, NavigationState> {
  parentRef: any;

  constructor(props: Record<string, unknown>) {
    super(props);
    this.parentRef = React.createRef();
    this.onResize = this.onResize.bind(this);
    this.state = { width: 320 };
  }

  onResize(width: any): void {
    this.setState({ width });
  }

  render(): JSX.Element {
    const { width } = this.state;
    return (
      <>
        {width <= 768
          ? (
            <DefaultCollapsableNav />
          ) : (
            <nav className="menu flex">
              <ul className="menu-nav">
                <li className="nav-item"><NavLink to="/daycare" className="nav-link" exact>Home</NavLink></li>
                <li className="nav-item"><NavLink to="/daycare/centers" className="nav-link">Our Centers</NavLink></li>
                <li className="nav-item"><AboutNav /></li>
                <li className="nav-item"><ProgramsNav /></li>
                <li className="nav-item"><DefaultParentsNav /></li>
                <li className="nav-item"><NavLink to="/daycare/news" className="nav-link">News</NavLink></li>
                <li className="nav-item"><NavLink to="/daycare/careers" className="nav-link">Careers</NavLink></li>
                <li className="nav-item"><NavLink to="/daycare/contact" className="nav-link">Contact Us</NavLink></li>
              </ul>
            </nav>
          )}
        <ReactResizeDetector handleWidth handleHeight onResize={this.onResize} targetDomEl={this.parentRef.current} />
      </>
    );
  }
}
