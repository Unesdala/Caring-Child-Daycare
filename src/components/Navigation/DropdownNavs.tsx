import { useState } from 'react';
import { Dropdown, DropdownItem, DropdownMenu, DropdownToggle } from 'reactstrap';
import { NavLink as RRNavLink } from 'react-router-dom';

const AboutNav = (): JSX.Element => {
  const [isOpen, setIsOpen] = useState(false);
  const toggle = () => setIsOpen(!isOpen);

  return (
    <Dropdown
      nav
      isOpen={isOpen} toggle={toggle}
    >
      <DropdownToggle
        caret
        nav
        className="no-hover-color"
      >
        About Us
      </DropdownToggle>
      <DropdownMenu>
        <DropdownItem to="/about" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          Our Philosophy
        </DropdownItem>
        <DropdownItem to="/about/caring-child-history" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          Our History
        </DropdownItem>
        <DropdownItem to="/about/curriculum" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          Our Curriculum
        </DropdownItem>
        <DropdownItem to="/about/faq" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          FAQ
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  );
};

const ProgramNav = (): JSX.Element => {
  const [isOpen, setIsOpen] = useState(false);
  const toggle = () => setIsOpen(!isOpen);

  return (
    <Dropdown
      nav
      isOpen={isOpen} toggle={toggle}
    >
      <DropdownToggle
        caret
        nav
        className="no-hover-color"
      >
        Programs
      </DropdownToggle>
      <DropdownMenu>
        <DropdownItem to="/programs/infants-and-toddlers" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          Infants &amp; Toddlers
        </DropdownItem>
        <DropdownItem to="/programs/twos" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          Twos
        </DropdownItem>
        <DropdownItem to="/programs/preschool" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          Preschool
        </DropdownItem>
        <DropdownItem to="/programs/private-pre-k" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          Private Pre-K
        </DropdownItem>
        <DropdownItem to="/programs/after-school" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          After School
        </DropdownItem>
        <DropdownItem to="/programs/summer-camp" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          Summer Camp
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  );
};

const LocationsNav = (): JSX.Element => {
  const [isOpen, setIsOpen] = useState(false);
  const toggle = () => setIsOpen(!isOpen);

  return (
    <Dropdown
      nav
      isOpen={isOpen} toggle={toggle}
    >
      <DropdownToggle
        caret
        nav
        className="no-hover-color"
      >
        Locations
      </DropdownToggle>
      <DropdownMenu>
        <DropdownItem to="/locations/vinton-center" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          Vinton Center
        </DropdownItem>
        <DropdownItem to="/locations/460-center" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          460 Center
        </DropdownItem>
        <DropdownItem to="/locations/parkway-center" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          Parkway Center
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  );
};

const ParentsNav = (): JSX.Element => {
  const [isOpen, setIsOpen] = useState(false);
  const toggle = () => setIsOpen(!isOpen);

  return (
    <Dropdown
      nav
      isOpen={isOpen} toggle={toggle}
    >
      <DropdownToggle
        caret
        nav
        className="no-hover-color"
      >
        The Parent's Corner
      </DropdownToggle>
      <DropdownMenu>
        <DropdownItem to="/parents/parents-corner" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          Parent's Corner
        </DropdownItem>
        <DropdownItem to="/parents/parents-resources" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          Parent's Resources
        </DropdownItem>
        <DropdownItem to="/parents/testimonials" exact activeClassName="active" tag={RRNavLink} onClick={toggle}>
          Testimonials
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  );
};

export { AboutNav, ProgramNav, LocationsNav, ParentsNav };