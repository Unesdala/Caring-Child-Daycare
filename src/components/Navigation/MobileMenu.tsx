import { useState } from 'react';
import { NavLink as RRNavLink } from 'react-router-dom';
import {
  Collapse, Navbar, NavbarToggler, NavLink,
} from 'reactstrap';
import 'bootstrap/dist/css/bootstrap.css';

const MobileNav = (): JSX.Element => {
  const [collapsed, setCollapsed] = useState(true);

  const toggleNavbar = () => setCollapsed(!collapsed);

  return (
    <>
      <Navbar light>
        <NavbarToggler onClick={toggleNavbar} />
        <Collapse isOpen={!collapsed} navbar>
          <ul className="menu-nav">
            <NavLink 
              to="/" 
              exact activeClassName="active" 
              tag={RRNavLink} 
              onClick={toggleNavbar}
            >
              Home
            </NavLink>
          </ul>
        </Collapse>
      </Navbar>
    </>
  );
};

export default MobileNav;