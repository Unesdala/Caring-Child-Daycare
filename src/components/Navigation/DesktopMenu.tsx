import { NavLink } from 'react-router-dom';
import { AboutNav, LocationsNav, ParentsNav, ProgramNav } from './DropdownNavs';

export const DesktopMenu = (): JSX.Element => {
  return (
    <>
      <nav className="menu">
        <ul className="menu-nav">
          <li className="nav-item">
            <NavLink to="/" className="nav-link" exact activeClassName="active">
              Home
            </NavLink>
          </li>
            <AboutNav />
            <ProgramNav />
            <LocationsNav />
            <ParentsNav />
          <li className="nav-item">
            <NavLink to="/careers/" className="nav-link" exact activeClassName="active">
              Careers
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink to="/contact/" className="nav-link" exact activeClassName="active">
              Contact Us
            </NavLink>
          </li>
        </ul>
      </nav>
    </>
  );
};

export default DesktopMenu;