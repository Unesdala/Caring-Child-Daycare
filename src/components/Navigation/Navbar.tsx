import React, { RefObject } from 'react';
import { Navbar } from 'react-bootstrap';
import { withResizeDetector } from 'react-resize-detector';
import { NavLink } from 'react-router-dom';
import { AboutNav, ProgramNav, LocationsNav, ParentsNav } from './DropdownNavs';

interface NavigationProps {
  targetRef: RefObject<HTMLDivElement>;
  width: number;
}

export class Navigation extends React.Component<NavigationProps> {

  render(): JSX.Element {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const { width, targetRef } = this.props;
    return (
      <div ref={targetRef} className="navigation">
        <Navbar collapseOnSelect expand="lg">
          <Navbar.Toggle aria-controls="responsive-navbar-nav" />
          <Navbar.Collapse id="responsive-navbar-nav">
            <nav className="menu">
              <ul className="menu-nav">
                <li className="nav-item">
                  <NavLink to="/" className="nav-link" exact activeClassName="active">
                    Home
                  </NavLink>
                </li>
                <AboutNav />
                <ProgramNav />
                <LocationsNav />
                <ParentsNav />
                <li className="nav-item">
                  <NavLink to="/careers/" className="nav-link" exact activeClassName="active">
                    Careers
                  </NavLink>
                </li>
                <li className="nav-item">
                  <NavLink to="/contact/" className="nav-link" exact activeClassName="active">
                    Contact Us
                  </NavLink>
                </li>
              </ul>
            </nav>
          </Navbar.Collapse>
        </Navbar>
      </div>
    );
  }
}

export default withResizeDetector(Navigation);