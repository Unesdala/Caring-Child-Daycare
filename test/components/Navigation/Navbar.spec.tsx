
import { shallow } from 'enzyme';
import { Navigation } from 'src/components/Navigation/Navbar';
import DesktopMenu from 'src/components/Navigation/DesktopMenu';
import MobileMenu from 'src/components/Navigation/MobileMenu';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const targetRef: any = {};
const wrapper = shallow<Navigation>(<Navigation targetRef={targetRef} width={1200} />);

describe('Navigation', () => {
  it('Renders the Navbar', () => {
    expect(wrapper.find(DesktopMenu).exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(DesktopMenu).dive().find('ul.menu-nav').exists()).toBe(true);
  });
  it('Renders the Navbar on cellphone width', () => {
    const wrapper2 = shallow<Navigation>(<Navigation targetRef={targetRef} width={320} />);
    expect(wrapper2.find(MobileMenu).exists()).toBe(true);
    expect(wrapper2).toMatchSnapshot();
  });
});