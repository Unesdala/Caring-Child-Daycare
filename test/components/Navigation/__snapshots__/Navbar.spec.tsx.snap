// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Renders the Navbar 1`] = `ShallowWrapper {}`;

exports[`Navigation Renders the Navbar on cellphone width 1`] = `ShallowWrapper {}`;
